mitosheet/src/mito/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -143,148 +125,179 @@ const GraphSidebar = (props: {
 
     }, [graphOutput])
 
-    /* 
-        This is the actual function responsible for loading the new
-        graph from the backend, making sure this graph is the correct
-        size.
-    */
-    const getGraphAsync = async () => {
-        // The reason that we use the mito-center-content-container instead of the graph-div is because the size of the graph div
-        // changes depending on the size of the graph. Specifically, when exiting fullscreen mode, the graph-div is wider
-        // than we actually have space for. 
-        const boundingRect: DOMRect | undefined = props.mitoContainerRef.current
-            ?.querySelector('#graph-div')
-            ?.getBoundingClientRect();
+    console.log("NEW SELECTED GRAPH", props.openGraph.graphID)
+
+    // If the number of graphs decreases, we make sure we're still on a graph that exists
+    // By switching either:
+    // 1. to a data sheet, if there are no graphs left
+    // 2. to the next graph, if we were on a graph that was deleted
+    // 3. to the last graph
+
+    const previousNumberOfGraphsRef = React.useRef<number>(props.graphDataArray.length);
+    const previouslySelectedGraphIndexRef = React.useRef<number>(props.graphDataArray.findIndex(graphData => graphData.graph_id === props.openGraph.graphID));
+    useEffect(() => {
+        const handleDeletedGraphs = async () => {
```

Comment:
Wouldn't we want to just do all of this handling when a graph is deleted? Instead of listening for the change of the length of `graphDataArray`?

Reply:
Per my comment below, there's just a few cases we have to handle. 

If you want to try and make that change, I'm down for it. Maybe:
1. create a `async deleteGraph` function that deletes the graph using the API, then switches to the right new graph tab (or no graph tab, if there is none)
2. Call it in the graph context menu, in the action, and in the modal that deletes graphs that rely on a dataset that get deleted